app/src/main/java/com/example/b07group7project/itempreview/ItemPreviewFragment.java

```diff
@@ -0,0 +1,98 @@
+package com.example.b07group7project.itempreview;
+
+import android.os.Bundle;
+
+import androidx.annotation.NonNull;
+import androidx.fragment.app.Fragment;
+
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.Button;
+import android.widget.ImageButton;
+import android.widget.TextView;
+
+import com.example.b07group7project.R;
+
+public class ItemPreviewFragment extends Fragment {
+    private TextView cartItemQtyTextView;
+    private StoreItem currentItem;
+    private int cartItemQty = 0;
+
```

Comment:
Instead of having the currentItem as a field of the fragment, you can can call the interface and have the interface give you the data for the currenItem. 

For example you can do something like currentItem = getItemInformation(...), you can look at the CheckoutFragment for an example of how this would kinda work.

You could also do what Aaron mentioned but I believe this would be a better way to implement the change because then we can just create the fragment without having to pass any parameters